cdc/changefeed.go

```diff
@@ -712,6 +724,32 @@ func (c *changeFeed) calcResolvedTs(ctx context.Context) error {
 		}
 	}
 
+	//sync-point on
+	if c.info.SyncPointEnabled {
+		c.syncpointMutex.Lock()
```

Comment:
Recommend adding `defer c.syncpointMutex.Unlock()` after this line, or the `Unlock` call should be carefully used.
Such as L757, L818 should call `c.syncpointMutex.Unlock()` because the function is terminated early